instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py

```diff
@@ -432,3 +434,25 @@ def test_tracer_provider_traced(self):
         self.assertEqual(
             span.resource.attributes["resource-key"], "resource-value"
         )
+
+    
+    def test_django_with_wsgi_instrumented(self):
+        """ 
+        This test is related to https://github.com/open-telemetry/opentelemetry-python-contrib/issues/448
+        Here I have added a parent span of kind SpanKind.SERVER in the context 
+        and on sending the request, the newly created span should be of type SpanKind.INTERNAL and 
+        it should be child to the parent_span
+        """
+
+        tracer = self.tracer_provider.get_tracer(__name__)
+        with tracer.start_as_current_span("test", kind=SpanKind.SERVER) as parent_span:
+            resp = Client().get("/span_name/1234/")
+            span_list = self.exporter.get_finished_spans()
+            print(span_list)
+            self.assertEqual(span_list[0].attributes[SpanAttributes.HTTP_STATUS_CODE], 200)
+            self.assertEqual(trace.SpanKind.INTERNAL, span_list[0].kind)
+            self.assertEqual(parent_span.get_span_context().span_id, span_list[0].parent.span_id)
+                
+            
+        
+
```

Comment:
I think you don't need this description anymore since you changed the implementation and issue #445 gives enough context.

```suggestion
        tracer = self.tracer_provider.get_tracer(__name__)
        with tracer.start_as_current_span("test", kind=SpanKind.SERVER) as parent_span:
            resp = Client().get("/span_name/1234/")
            span_list = self.exporter.get_finished_spans()
            self.assertEqual(span_list[0].attributes[SpanAttributes.HTTP_STATUS_CODE], 200)
            self.assertEqual(trace.SpanKind.INTERNAL, span_list[0].kind)
            self.assertEqual(parent_span.get_span_context().span_id, span_list[0].parent.span_id)
```

Reply:
@lonewolf3739 : Done